### PR TITLE
Check API version before calling `isCacheEnabled`

### LINKED ISSUE #6598: Incremental PR analysis fails for SQ < 9.4
### Description

When doing incremental PR analysis on a SonarQube version older than 9.4 the end step fails with:
```
java.lang.NoSuchMethodError: 'boolean org.sonar.api.batch.sensor.SensorContext.isCacheEnabled()'
	at org.sonarsource.dotnet.shared.plugins.AbstractFileCacheSensor.execute(AbstractFileCacheSensor.java:58)
	at org.sonar.scanner.sensor.AbstractSensorWrapper.analyse(AbstractSensorWrapper.java:48)
	at org.sonar.scanner.sensor.ProjectSensorsExecutor.execute(ProjectSensorsExecutor.java:49)
	at org.sonar.scanner.scan.ProjectScanContainer.doAfterStart(ProjectScanContainer.java:361)
	at org.sonar.core.platform.ComponentContainer.startComponents(ComponentContainer.java:136)
	at org.sonar.core.platform.ComponentContainer.execute(ComponentContainer.java:122)
	at org.sonar.scanner.bootstrap.GlobalContainer.doAfterStart(GlobalContainer.java:141)
	at org.sonar.core.platform.ComponentContainer.startComponents(ComponentContainer.java:136)
	at org.sonar.core.platform.ComponentContainer.execute(ComponentContainer.java:122)
	at org.sonar.batch.bootstrapper.Batch.doExecute(Batch.java:73)
	at org.sonar.batch.bootstrapper.Batch.execute(Batch.java:67)
	at org.sonarsource.scanner.api.internal.batch.BatchIsolatedLauncher.execute(BatchIsolatedLauncher.java:46)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.sonarsource.scanner.api.internal.IsolatedLauncherProxy.invoke(IsolatedLauncherProxy.java:60)
	at com.sun.proxy.$Proxy0.execute(Unknown Source)
	at org.sonarsource.scanner.api.EmbeddedScanner.doExecute(EmbeddedScanner.java:189)
	at org.sonarsource.scanner.api.EmbeddedScanner.execute(EmbeddedScanner.java:138)
	at org.sonarsource.scanner.cli.Main.execute(Main.java:112)
	at org.sonarsource.scanner.cli.Main.execute(Main.java:75)
	at org.sonarsource.scanner.cli.Main.main(Main.java:61)
```

The problem appears because the `isCacheEnabled` method was added in 9.4 and there is no backward compatibility support for this.

The impact of this issue should be very limited since the plugins are delivered as part of SonarQube and users are impacted only if are manually updating the plugin on an older version of SonarQube (<9.4).

### PR DESCRIPTION
**Pull request by costin-zaharia-sonarsource**
Fixes #6598

I've validated this locally by updating the integration tests to run with 8.9. The problem is reproducible before the change and is fixed after it.

### CODE REVIEW

File: sonar-dotnet-shared-library/src/main/java/org/sonarsource/dotnet/shared/plugins/AbstractFileCacheSensor.java [hunk 1]
```diff
@@ -55,6 +59,11 @@ public void execute(SensorContext context) {
       return;
     }
 
+    if (!runtime.getApiVersion().isGreaterThanOrEqual(Version.create(9, 4))){
```
Comment (martin-strecker-sonarsource) [inline sonar-dotnet-shared-library/src/main/java/org/sonarsource/dotnet/shared/plugins/AbstractFileCacheSensor.java:62]: Is there also a `getServerVersion()`? If so, we need to make sure in the tests, that we call the right version check.

**Review by martin-strecker-sonarsource — COMMENTED** [review]
I have some questions that might result in some more work to be done.

File: sonar-dotnet-shared-library/src/test/java/org/sonarsource/dotnet/shared/plugins/AbstractFileCacheSensorTest.java [hunk 2]
```diff
@@ -45,6 +50,8 @@
 public class AbstractFileCacheSensorTest {
   private static final String LANGUAGE_KEY = "language-key";
   private static final String LANGUAGE_NAME = "Language Name";
+  private static final SonarRuntime RUNTIME_WITH_ANALYSIS_CACHE = SonarRuntimeImpl.forSonarQube(Version.create(9, 4), SonarQubeSide.SERVER, SonarEdition.COMMUNITY);
```
[the remark this answers is not in the document]
Comment (costin-zaharia-sonarsource) [reply sonar-dotnet-shared-library/src/test/java/org/sonarsource/dotnet/shared/plugins/AbstractFileCacheSensorTest.java:53]: You can only specify the api version.

File: sonar-dotnet-shared-library/src/main/java/org/sonarsource/dotnet/shared/plugins/AbstractFileCacheSensor.java (hunk 1 above)
Comment (costin-zaharia-sonarsource) [reply sonar-dotnet-shared-library/src/main/java/org/sonarsource/dotnet/shared/plugins/AbstractFileCacheSensor.java:62]: No, as far as I can tell you cannot ask for the server version.

**Review by martin-strecker-sonarsource — APPROVED** [review]
LGTM. I did not merged it because you declared master frozen. You may unfreeze it also.

**Comment by costin-zaharia-sonarsource** [comment]
> I did not merged it because you declared master frozen. You may unfreeze it also.

I thought that it was clear from the discussion thread that we will not do the bugfix release anymore but sure, I can add a comment to be clear.

**Pull request merged**